### PR TITLE
Update Docs for v0.11 release

### PR DESCRIPTION
**Pull request by jeremiahpslewis**
PR Checklist

- [ ] Update NEWS.md?
- [ ] Unit tests for all structs / functions?
- [ ] Integration and correctness tests using a simple env?
- [ ] PR Review?
- [ ] Add or update documentation?
- [ ] Write docstrings for new methods?

### CODE REVIEW

**Comment by jeremiahpslewis** [comment]
@HenriDeh after this PR and https://github.com/JuliaReinforcementLearning/ReinforcementLearning.jl/pull/1055 have been merged, I'm ready to tag v0.11. Any objections?

**Comment by jeremiahpslewis** [comment]
@HenriDeh Ready to go (I hope)! Had to add three new objects `AbstractPlayer`, `Player`, `PlayerTuple` to deal with type ambiguities between actions / states and players, when actions / states use `Symbol` type.

File: docs/src/How_to_implement_a_new_algorithm.md [hunk 1]
```diff
@@ -10,43 +10,49 @@ function _run(policy::AbstractPolicy,
         stop_condition::AbstractStopCondition,
         hook::AbstractHook,
         reset_condition::AbstractResetCondition)
-
+    push!(hook, PreExperimentStage(), policy, env)
```
Comment (HenriDeh) [inline docs/src/How_to_implement_a_new_algorithm.md:13]: If you add the hooks in this loop, then you should remove the "hooks were removed ..." above.

File: docs/src/How_to_use_hooks.md [hunk 2]
```diff
@@ -120,37 +124,30 @@ run(
 [BSON.jl](https://github.com/JuliaIO/BSON.jl) is recommended to save the parameters of a policy.
```
Comment (HenriDeh) [inline docs/src/How_to_use_hooks.md:124]: Flux suggests JLD2 as well. I personnaly encountered issues with BSON when working accross different julia versions.

File: docs/src/How_to_write_a_customized_environment.md [hunk 3]
```diff
@@ -102,12 +109,12 @@ RLBase.test_runnable!(env)
...
 It is a simple smell test which works like this:
 
-```
+```julia
 n_episode = 10
 for _ in 1:n_episode
     reset!(env)
     while !is_terminated(env)
-        env |> action_space |> rand |> env
+        env |> action_space |> rand |> (x -> act!(env, x))
```
Comment (HenriDeh) [inline docs/src/How_to_write_a_customized_environment.md:117]:
```suggestion
        action = rand(action_space(env)) 
        act!(env, action)
```
This is clearer imo.

File: docs/src/tips.md [hunk 4]
```diff
@@ -2,32 +2,22 @@
...
+Pkg.develop(path="src/ReinforcementLearningBase")
+Pkg.develop(path="src/ReinforcementLearningCore")
+Pkg.develop(path="src/ReinforcementLearningEnvironments")
+Pkg.develop(path="src/ReinforcementLearningFarm") # optional
+```
+
 Sometimes, you may need to add some
 extra dependencies. Remember to switch the environment before adding new
 packages. For example, if you want to add
 `Statistics` in `ReinforcementLearningBase`, first run `]activate
```
Comment (HenriDeh) [inline docs/src/tips.md:18]:
```suggestion
`Statistics` to `ReinforcementLearningBase`, first run `]activate
```

**Review by HenriDeh — COMMENTED** [review]
a few remarks otherwise great job !

**Comment by jeremiahpslewis** [comment]
One more thing...`ComposedStop` was creating all kinds of type stability issues, for little gain; it takes 4-5 lines to define a custom `ComposedStop` with your favorite reducer. So I dropped it, we now have `StopIfAll` and `StopIfAny`; if there's a third really popular one, it can go in the 🐮 🐷 🐔 Farm, but for now we cover 90% of typical usage. ;) @HenriDeh

**Comment by jeremiahpslewis** [comment]
@HenriDeh Thanks for the thorough review and helpful comments, the PR got a bit out of hand towards the end!

**Review by HenriDeh — APPROVED** [review]

**Pull request merged**